[DOCS] Update remote cluster docs (#77043)
elastic/elasticsearch: docs/reference/modules/cluster/remote-clusters-security.asciidoc

[[remote-clusters-security]]
=== Configure remote clusters with security
To use {ccr} or {ccs} with secured clusters, all clusters must have the {es} 

> security features enabled.

Author: I made some bigger changes about explaining security impacts in the main remote clusters page, but this language needs updating as well. I think that we want to inform users that security is really important with CCR/CCS, though it isn't actually _required_ (to my knowledge). That said, I think we can change this to read:
>To use {ccr} or {ccs} safely with remote clusters, enable security on all connected clusters and configure Transport Layer Security (TLS) on every node.

> security features enabled.

Reviewer: Yes security is not required for remote clusters since CCS is a core feature. But we should strong recommend it. And also be explicit about "secured" means both configuring TLS (at least for transport) and RBAC (both requires `xpack.security.enabled: true`)

The local cluster must be allowed to connect to the remote clusters, which means 
that the certificate authority (CA) used to sign the keystore of the local

> cluster must be trusted by the remote clusters.

Author: That's a better way of explaining! How about this revision?
>All connected clusters must trust one another and be mutually authenticated at the TLS level. This means that the local cluster trusts the certificate authority (CA) of the remote cluster, and the remote cluster trusts the CA of
the local cluster.

> cluster must be trusted by the remote clusters.

Reviewer: Yes that reads better. Please also see my [other comment](https://github.com/elastic/elasticsearch/pull/77043#discussion_r707839179) for more details about mutual trust.

> cluster must be trusted by the remote clusters.

Author: Thanks for that explanation @ywangd -- it's very helpful. Based on your other comment, I'm expanding this section a bit: 
>All connected clusters must trust one another and be mutually authenticated at the TLS level. This means that the local cluster trusts the certificate authority (CA) of the remote cluster, and the remote cluster trusts the CA of
the local cluster. When establishing a connection, all nodes will verify certificates from the nodes they are connecting to. This mutual trust is required to connect a remote cluster, because all connected nodes effectively form a single security domain.

User authentication is performed on the local cluster and the user and user’s 
roles are passed to the remote clusters. A remote cluster checks the user’s
roles against its local role definitions to determine which indices the user is 
allowed to access.

Before using {ccr} or {ccs} with secure {es} clusters, complete the following 
configuration tasks:

. Enable the {es} {security-features} on every node in each connected cluster by
setting `xpack.security.enabled` to `true`. Refer to the
<<general-security-settings,{es} security settings>>.

. Encrypt internode communications with Transport Layer Security (TLS) on every
node to establish a trust relationship between the local cluster and all remote

> clusters. Refer to 

Author: You're right, it doesn't read correctly. Does this revision better explain?
>Configure Transport Layer Security (TLS) on every node to encrypt internode traffic and authenticate nodes in the local cluster with nodes in all remote clusters.

<<security-basic-setup,set up basic security for the {stack}>> for the required
steps to configure security.
+
NOTE: This procedure uses the same CA to generate certificates for all nodes.
Alternatively, you can add the CA certificates from the local cluster as a

> trusted CA in each remote cluster. 

Author: Hmm...interesting. Connections to a remote cluster are typically unidirectional, correct? Meaning you connect to remote clusters from a local cluster. But if a remote cluster trusts its CA certificates, and those are trusted by the local cluster, then all connected clusters (both local and remote) are trusting the same certificates. We should test both scenarios so that users are aware.

> trusted CA in each remote cluster. 

Reviewer: > Connections to a remote cluster are typically unidirectional, correct?

It appears to be unidirectional because the conceptual data flow is one way (especially for CCR). But cross cluster communication uses the same Elasticsearch Transprot layer protocol. This means if the transport layer is configured to require mutual TLS authentication (and it should based on our recommended best practice), remote and local nodes are treated no differently at the transport layer. When establishing conneciton, each  party will verify certificate from the other side. Thus they require trust each other's certificate or CA.

You can essentially picture those nodes (local and remote) forming a single cluster at transport layer. Indeed, CCS effectively works that way, i.e. it treats shards on the remote nodes the same way as local shards.

In summary, setting up remote cluster really means full mutual trust. They are essentially the same security domain. There were questions around whether we can separate them at both the TLS and RABC layers. I think we want to make improvements. But there has not been any formal planning yet.

. Connect <<remote-clusters-connect,remote clusters>> to a local cluster.

. Configure <<remote-clusters-privileges,remote cluster privileges>>.

[[remote-clusters-security-ccr]]
==== {ccr-cap} and security
After enabling and configuring security, connecting a remote cluster, and
configuring privileges, you need to
<<ccr-getting-started-follower-index,configure a follower index>> on your local

> cluster to replicate on a remote cluster.

Author: I only called out CCR because I didn't think that any additional steps were required for CCS. It does feel odd. I just wasn't sure what to add here. We could add some information like, "You can now [search across](https://www.elastic.co/guide/en/elasticsearch/reference/master/modules-cross-cluster-search.html) single or multiple remote clusters.", though that feels a little weak.

[[remote-clusters-security]]
=== Configure remote clusters with security
To use {ccr} or {ccs} safely with remote clusters, enable security on all 

> connected clusters and configure Transport Layer Security (TLS) on every node.

Author: Let's add:
>Configuring TLS security at the transport level is minimally required for remote clusters. For additional security, configure TLS at the <<security-basic-setup-https,HTTP level>> as well.

Then in the next sentence, we can say:
>All connected clusters must trust one another and be mutually authenticated at the TLS level on at least the transport layer.

certificates from the nodes they are connecting to. This mutual trust is
required to securely connect a remote cluster, because all connected nodes
effectively form a single security domain.

User authentication is performed on the local cluster and the user and user’s 
roles names are passed to the remote clusters. A remote cluster checks the user’s
roles against its local role definitions to determine which indices the user is 
allowed to access.

Before using {ccr} or {ccs} with secure {es} clusters, complete the following 
configuration tasks:

. Enable the {es} {security-features} on every node in each connected cluster by
setting `xpack.security.enabled` to `true`. Refer to the
<<general-security-settings,{es} security settings>>.

. Configure Transport Layer Security (TLS) on every node to encrypt internode
traffic and authenticate nodes in the local cluster with nodes in all remote
clusters. Refer to 
<<security-basic-setup,set up basic security for the {stack}>> for the required
steps to configure security.
+
NOTE: This procedure uses the same CA to generate certificates for all nodes.
Alternatively, you can add the CA certificates from the local cluster as a

> trusted CA in each remote cluster. The opposite is also true.

Author: Ah, so both are required? Meaning you must add the CA certificates from the local cluster as a trusted CA in each remote cluster, and _also_ add the CA certificate from the remote clusters to the local cluster? If that's true, we should mention that using the same CA on all remote clusters would simplify this task.

After enabling and configuring security, you can 
<<remote-clusters-connect,connect remote clusters>> from a local cluster.

With your clusters connected, you'll need to 
<<remote-clusters-privileges,configure cluster privileges>> on both the local
and remote clusters.

If you're configuring a remote cluster for {ccr}, you need to 
<<ccr-getting-started-follower-index,configure a follower index>> on your local

> cluster to replicate on a remote cluster.

Author: I missed this change when merging in other changes, but I'll include it in the next push.

[[remote-clusters-security]]
=== Configure remote clusters with security
To use {ccr} or {ccs} safely with remote clusters, enable security on all 
connected clusters and configure Transport Layer Security (TLS) on every node.
Configuring TLS security at the transport level is minimally required for
remote clusters. For additional security, configure TLS at the
<<security-basic-setup-https,HTTP level>> as well.

All connected clusters must trust one another and be mutually authenticated at

> the TLS level on at least the transport layer. This means that the local cluster

Reviewer: ```suggestion
with TLS on the transport interface. This means that the local cluster
```

We can drop "at least" since cross cluster does not use the HTTP level.

trusts the certificate  authority (CA) of the remote cluster, and the remote
cluster trusts the CA of the local cluster. When establishing a connection, all

> nodes will verify certificates from the nodes they are connecting to. This 

Reviewer: ```suggestion
nodes will verify certificates from the nodes on the other side. This 
```

It is not just "connecting to". It also happens for "connecting from" if that makes sense. Basically the verfication is mutual. regardless which side initiates it.

mutual trust is required to securely connect a remote cluster, because all 
connected nodes effectively form a single security domain.

User authentication is performed on the local cluster and the user and user’s 
roles names are passed to the remote clusters. A remote cluster checks the user’s

> roles against its local role definitions to determine which indices the user is 

Reviewer: ```suggestion
role names against its local role definitions to determine which indices the user is 
```

allowed to access.

Before using {ccr} or {ccs} with secured {es} clusters, complete the following 
configuration tasks:

. Enable the {es} {security-features} on every node in each connected cluster by
setting `xpack.security.enabled` to `true`. Refer to the
<<general-security-settings,{es} security settings>>.

. Configure Transport Layer Security (TLS) on every node to encrypt internode
traffic and authenticate nodes in the local cluster with nodes in all remote
clusters. Refer to 
<<security-basic-setup,set up basic security for the {stack}>> for the required
steps to configure security.
+
NOTE: This procedure uses the same CA to generate certificates for all nodes.
Alternatively, you can add the certificates from the local cluster as a
trusted CA in each remote cluster. You must also add the certificates from
remote clusters as a trusted CA on the local cluster. Using the same CA to
generate certificates for all nodes simplifies this task.

After enabling and configuring security, you can 
<<remote-clusters-connect,connect remote clusters>> from a local cluster.

With your clusters connected, you'll need to 

> <<remote-clusters-privileges,configure cluster privileges>> on both the local

Reviewer: ```suggestion
<<remote-clusters-privileges,configure users and privileges>> on both the local
```
